context-free.md

```diff
@@ -2,21 +2,33 @@
 
 ##  Who is this programming language for?
 
+I think the ideal target audience would be a computer scientist who's interested in art. This is because the output of the program can be very artistic in nature, which appeals to artists, but knowing about context free grammars (which most computer scientists do) enables one to understand how the language works and thus pick it up quickly.  
 
 ## What is easy to do in this language? Why is it easy?
 
+It's easy to create images using recursion, like fractals. This is easy because ContextFree programs are (unsurprisingly) structured like the specification of a context-free grammar, with a start symbol that can call other rules recursively. 
 
 ## What is hard to do in this language? Why is it hard?
 
+It's hard to specify complex shapes that aren't recursively defined. This is hard because you'd have to actually specify the shape line by line - consider, for example, the amount of code needed to create the "Welcome" image. It's also hard to visualize what your programs are going to look like without actually compiling them, because it's hard to visualize what the recursion will look like. 
```

Comment:
Good comment on visualization. That's something to consider when writing a DSL. Is render on the fly possible?